tests/test_amend.py

```diff
@@ -179,6 +179,68 @@ def test_author_set_supplier_components(self) -> None:
             self.sbom_fixture["components"][0]["author"],
         )
 
+    def test_supplier_gets_not_overwritten(self) -> None:
```

Comment:
what is the difference between this test and `test_supplier_name_url_get_no_overwritten`?